mParticle-iterable/MPKitIterable.m

```diff
@@ -62,10 +89,137 @@ - (nonnull MPKitExecStatus *)continueUserActivity:(nonnull NSUserActivity *)user
         
         [self->_kitApi onAttributionCompleteWithResult:attributionResult error:nil];
     };
+
     [IterableAPI getAndTrackDeeplink:clickedURL callbackBlock:callbackBlock];
-    
+
     MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:[MPKitIterable kitCode] returnCode:MPKitReturnCodeSuccess];
     return execStatus;
 }
 
+- (void)initIntegrationAttributes {
+    NSDictionary *integrationAttributes = @{
+            @"Iterable.sdkVersion": IterableAPI.sdkVersion
+    };
+    [[MParticle sharedInstance] setIntegrationAttributes:integrationAttributes forKit:MPKitIterable.kitCode];
+}
+
+
+- (MPKitExecStatus *)onUserIdentified:(NSNotification*) notification {
+    FilteredMParticleUser *user = [self currentUser];
+    [self updateIdentity:user];
+    return [[MPKitExecStatus alloc] initWithSDKCode:[[self class] kitCode] returnCode:MPKitReturnCodeSuccess];
+}
+
+- (NSString *)getUserEmail:(FilteredMParticleUser *)user {
+    return user.userIdentities[@(MPUserIdentityEmail)];
+}
+
+- (NSString *)getCustomerId:(FilteredMParticleUser *)user {
+    return user.userIdentities[@(MPUserIdentityCustomerId)];
+}
+
+- (MPKitExecStatus *)onLoginComplete:(FilteredMParticleUser *)user request:(FilteredMPIdentityApiRequest *)request {
+    [self updateIdentity:user];
+    return [[MPKitExecStatus alloc] initWithSDKCode:[[self class] kitCode] returnCode:MPKitReturnCodeSuccess];
+}
+
+- (nonnull MPKitExecStatus *)onLogoutComplete:(nonnull FilteredMParticleUser *)user request:(nonnull FilteredMPIdentityApiRequest *)request {
+    [self updateIdentity:user];
+    return [[MPKitExecStatus alloc] initWithSDKCode:[[self class] kitCode] returnCode:MPKitReturnCodeSuccess];
+}
+
+- (MPKitExecStatus *)onIdentifyComplete:(FilteredMParticleUser *)user request:(FilteredMPIdentityApiRequest *)request {
+    [self updateIdentity:user];
+    return [[MPKitExecStatus alloc] initWithSDKCode:[[self class] kitCode] returnCode:MPKitReturnCodeSuccess];
+}
+
+- (NSString *)getPlaceholderEmail:(FilteredMParticleUser *)user {
+    NSString *id = nil;
+    if (self.mpidEnabled) {
+        if (user.userId.longValue != 0) {
+            id = user.userId.stringValue;
+        }
+    } else {
+        id = [[[UIDevice currentDevice] identifierForVendor] UUIDString];
+
+        if (!id.length) {
+            id = [self advertiserId];
```

Comment:
I think on the server-side this may end up being all lowercase, but here will be capital. to maintain consistency you may need to lowercase this.

Reply:
Where is it lowercased? I don't see that in the mParticle SDK or lambda code..

Reply:
Done